SDWebImage/Private/SDInternalMacros.h

```diff
@@ -7,14 +7,28 @@
  */
 
 #import <Foundation/Foundation.h>
+#import <os/lock.h>
+#import <libkern/OSAtomic.h>
 #import "SDmetamacros.h"
 
+#ifndef SD_LOCK_DECLARE
+#define SD_LOCK_DECLARE(lock) os_unfair_lock lock API_AVAILABLE(ios(10.0), tvos(10), watchos(3), macos(10.12)); \
+OSSpinLock lock##_deprecated;
+#endif
```

Comment:
`OSSpinLock lock##_deprecated` Does this need to judge whether it is declared below ios10 ?

Reply:
Declare, can not use platform check. Because, it's not a runtime behavior. It's compiler's compile behavior.

Reply:
How about like this ?

```objc
#if __IPHONE_OS_VERSION_MIN_REQUIRED < 100000
OSSpinLock lock##_deprecated;
#endif
```

Reply:
This is SDK version, not `current running iOS version on my iPhone`

Reply:
Maybe we can upgrade the minimum dependency version of SDK to ios10 after the version of 6.0.